inc/datahandlers/user.php

```diff
@@ -1536,17 +1536,24 @@ function delete_content($delete_uids=false)
 		$plugins->run_hooks('datahandler_user_delete_content', $this);
 
 		$db->delete_query('userfields', "ufid IN({$this->delete_uids})");
-		$db->delete_query('privatemessages', "uid IN({$this->delete_uids})");
+		$db->delete_query('privatemessages', "uid IN({$this->delete_uids}) OR fromid IN({$this->delete_uids})");
```

Comment:
@Destroy666x While purge spammer should delete PMs other users received this method is also used by the user pruning task where PMs shouldn't be deleted.